### DISCUSSION TITLE
What are the drawbacks of using supabase with directus?

### DISCUSSION THREAD

**Discussion by Nisthar** [directus/directus#17444] [Archive]
I have an existing directus hosted using docker. There are some users and database records. 

How do i add supabase on to it? Does this mean directus roles become unusable? 

Can this process be reversed in the future?

**Comment by Nisthar** [comment]
The upsides would be the auth libraries, realtime, etc

**Reply by prodkt** [reply]
Not really. Supabase has its own users bank, and that is separate from the directus Table. If you want to use supabase for auth and users some sort of remapping of directus users and roles and permissions would need to happen.

**Comment by rijkvanzanten** [comment]
Hopefully https://supabase.com/docs/guides/integrations/directus and https://directus.io/blog/how-directus-and-supabase-are-partnering-to-support-backend-developers/ answer those questions 🙂

**Reply by paescuj** [reply]
I guess only if you want to have a single source for users, otherwise I think it should just work.

**Reply by Nisthar** [reply]
> Hopefully https://supabase.com/docs/guides/integrations/directus and https://directus.io/blog/how-directus-and-supabase-are-partnering-to-support-backend-developers/ answer those questions 🙂

Do you know how to connect supabase to a directus instance? I already have a directus instance. I would like to connect supabase to it.

**Reply by paescuj** [reply]
What exactly do you want to connect? What do you want to achieve with that?

**Reply by Nisthar** [reply]
> What exactly do you want to connect? What do you want to achieve with that?

I already have a directus instance. I want to connect supabase for the twitch signin support.

**Reply by development-vargamarcel** [reply]
As far as i know it doesn't work like that, meaning: If you want to use supabase's twitch signin you will have to use supabase's api. You cannot use a supabase server feature from directus api as is. 
WHAT YOU CAN DO: 
You can use one database for both.(this includes just that, the database.No unified auth, no unified permissions or file storage... none of that... Unless you integrate them if both give you that flexibility. But out of the box you get just that. )
 
**I tried to do some things like this with nhost and supabase and it isn't worth the trouble cause you have to always be make sure an update won't ruin your work.**

**Reply by raf-avag** [reply]
Is it sensible to set-up custom routes for API access which uses Supabase server features [basically ignoring the routes autogenerated by Directus] in this case? I'm pretty new to all this, and all I really need from Supabase is:

1. Extensions management, especially vector extension [which seems to be the easiest to use in other functions and setup]
2. Auth functionality (which Directus has as well, but is Supabase's better/worse??)
3. RLS (I believe it should be covered by Directus but haven't tried it, any idea?)
4. and Edge functions

> Unless you integrate them if both give you that flexibility

And does this mean customizing Directus' middleware, or are there other options??
